eos-updater-flatpak-installer/eos-updater-autoinstall.schema.json

```diff
@@ -88,6 +88,38 @@
                 }
             },
             "required": ["ref-kind", "name", "branch"]
+        },
+        {
+            "properties": {
+                "action": {
+                    "const": "prune-ref",
```

Comment:
@pwithnall would know better, but I am not sure that you can cleanly add new actions like this in a backwards-compatible way without an OS checkpoint. The actions from the OS update being deployed are parsed and prepared by the **old** eos-updater version.

I copied the example from the manpage to my local system, then did a dry run:

```
wjt@camille:~$ cat /var/lib/eos-application-tools/flatpak-autoinstall.d/99-prune-ref.json 
[
    {
        "action": "prune-ref",
        "branch": "stable",
        "serial": 2017100300,
        "ref-kind": "app",
        "name": "org.example.UninstalledApp",
        "collection-id": "org.example.Apps",
        "remote": "example-apps"
    }
]
wjt@camille:~$ /usr/libexec/eos-updater-flatpak-installer --dry-run --mode check 
eos-updater-flatpak-installer-Message: 09:21:29.098: Running in mode ‘check’

(eos-updater-flatpak-installer:4989): libeos-updater-util-WARNING **: 09:21:29.099: Skipping the following actions while parsing ‘99-prune-ref.json’, due to not supporting their contents. They will not be reapplied later; the system may be in an inconsistent state from this point forward.
{"action":"prune-ref","branch":"stable","serial":2017100300,"ref-kind":"app","name":"org.example.UninstalledApp","collection-id":"org.example.Apps","remote":"example-apps"}
[...]
```

I then tried installing an OS update with this file still in place. To my surprise the OS update was deployed despite this warning:

```
Dec 15 09:23:12 camille eos-updater[1915]: Skipping the following actions while parsing ‘99-prune-ref.json’, due to not supporting their contents. They will not be reapplied later; the system may be in an inconsistent state from this point forward.
                                           {"action":"prune-ref","branch":"stable","serial":2017100300,"ref-kind":"app","name":"org.example.UninstalledApp","collection-id":"org.example.Apps","remote":"example-apps"}
```

So it looks like I am wrong and in fact the action would be applied after booting into the new OS version which supports this feature.

Reply:
You might still be right (there's a big comment in the code about it), but this action isn't used in the pre-upgrade part by eos-updater since there's no fetching involved.

Reply:
I think we would need to thoroughly test whether an unrecognised action at the prepare phase would prevent:

- subsequent actions in the same file being prepared (I think yes)
- actions in subsequently-named files being prepared (?)

Reply:
I can’t remember what the code was meant to do (sorry), so defer to the big comment and (more importantly) testing of whether this actually works.